### PR TITLE
plugins/telesync

### PR DESCRIPTION
**Pull request by makiftasova**
A sync plugin between Hangouts and Telegram (with proper photo sync support)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,49 @@ aiohttp
 /hangupsbot/__pycache__
 /build
 /scripts
+.idea
+
+# telesync could keep photos, and probably we don't want that photos on git repo
+*.jpg 
+
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio and Webstorm
```
Comment (nylonee) [inline .gitignore]: Do we need all this in the .gitignore?

File: requirements.txt [hunk 2]
```diff
@@ -2,6 +2,7 @@
 git+https://github.com/hangoutsbot/hangups.git
 appdirs
 asyncio
+aiohttp >= 0.21.2
```
Comment (nylonee) [inline requirements.txt]: Will this affect other plugins or the core bot functionality?

File: hangupsbot/plugins/telesync.py [hunk 3]
```diff
@@ -0,0 +1,483 @@
...
+    bot.memory.set_by_path(['telesync_ho2tg'], ho2tg_dict)
+
+
+def is_valid_image_link(url):
+    if ' ' in url:
+        return False
+
+    url = url.lower()
+    if url.startswith('http://') or url.startswith('https://'):
+        if url.endswith((".jpg", ".gif", ".gifv", ".webm", ".png", ".mp4")):
```
Comment (nylonee) [inline hangupsbot/plugins/telesync.py]: .jpeg?

**Comment by nylonee** [comment]
Nice work!

I'm wondering if you have time to write some documentation for how to use this plugin, in the wiki.

File: requirements.txt (hunk 2 above)
Comment (makiftasova) [inline requirements.txt]: Photo syncs depends on aiohttp, afaik it should be at least aiohttp v0.20.2 in order to download photos and then upload photos with bot APIs

File: .gitignore (hunk 1 above)
Comment (makiftasova) [inline .gitignore]: I just wanted to keep my IDE configuration to myself. It shouldn't be required for bot functionality etc but I am a lazy guy to learn how to exclude files from pull requests

**Comment by makiftasova** [comment]
I have some time, but need to learn how to edit wiki first :)

**Pull request merged**